### PR TITLE
Bluetooth: Mesh: Model start callback

### PR DESCRIPTION
**Pull request by trond-snekvik**
Replaces the Mesh model settings_commit callback with a start callback,
indicating that the mesh model behavior is ready to start. Everything
that was previously done in the settings_commit callback may be moved to
this callback, which gets called just after mesh settings are committed,
instead of in the middle of the process.

This resolves an issue where models had no context in which to start
their behavior, as the previous settings_commit call fired before the
mesh was declared valid, making access APIs inaccessible.

Signed-off-by: Trond Einar Snekvik <Trond.Einar.Snekvik@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/mesh/net.c [hunk 1]
```diff
@@ -1407,6 +1407,7 @@ void bt_mesh_net_start(void)
 
 		bt_mesh_prov_complete(net_idx, addr);
 	}
+
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/net.c]: Seems like an unrelated addition?

File: include/bluetooth/mesh/access.h [hunk 2]
```diff
@@ -467,18 +467,22 @@ struct bt_mesh_model_cb {
...
 	 *
-	 *  @sa settings_handler::h_commit
+	 *  When this callback fires, the mesh model may start its behavior,
+	 *  and all Access APIs are ready for use.
 	 *
-	 *  @param model Model this callback belongs to.
+	 *  @param model      Model this callback belongs to.
+	 *  @param first_time Flag indicating whether this is the first time the
+	 *                    node is started. If false, provisioning data was
+	 *                    loaded from persistent storage.
```
Comment (jhedberg) [inline include/bluetooth/mesh/access.h]: What kind of use cases do you have in mind for this parameter? I.e. what kind of things could the callback decide (or not decide) to do with it?

Comment (trond-snekvik) [reply include/bluetooth/mesh/access.h]: To be honest, I don't have a strong use case for this at the moment. It makes it possible to retain any previous logic in the settings_commit callback, like reconstruction of state based on loaded values (as opposed to using default values the first time around), but I'm not sure if this is problem anyone would be facing. 

I added it because it's potentially interesting information that's otherwise lost after this callback. If you think it just complicates the API or if it's inaccurate somehow, I'm not going to fight for it.

**Comment by LingoMeng** [comment]
I think this approach is more natural in handling the OnPower behavior specified by the mesh model.

**Comment by LingoMeng** [comment]
@trond-snekvik  There are limitations in using the current model storage mechanism, for example, if you want to store two different kinds of data, you must encapsulate them in one structure, and store another or another member `repeatedly `when one member changes. I think it's best to store this data in the form of key value pairs.

**Comment by trond-snekvik** [comment]
@LingoMeng You make a good point, but this probably isn't the right place to discuss it. Could you open a separate issue for this?

**Review by jhedberg — APPROVED** [review]
